### ISSUE TITLE
buildMapping broken

### ISSUE THREAD

**Issue by mrasquin** [SCOREC/core#22]
When buildMapping is on in adapt.inp, Chef exits from core/apf/apf.cc #57 at the instruction "assert( ! m->findField(name));". 

So far, the adopted rule was to reconstruct the mapping  between the input and output mesh for each partitioning. Therefore Chef should not look for the mapping field in the phasta restart files.

**Comment by ibaned** [comment]
So... the situation you have is that your restart files have a mapping field, but you want Chef to ignore that and build a new mapping from scratch as usual ?

**Comment by mrasquin** [comment]
Correct. 
When I partition a mesh to a large part count, I try to limit the split factor and I create intermediate partitions for that purpose
If I take the example of one intermediate partition P2 between the initial partition P1 and the final partition P3, Chef used to produce a mapping between P1 and P2 during the first mesh split, which was stored in the resulting restart files associated with P2. Then, during the second mesh split, Chef used to produce a second mapping between P2 and P3 and save it in the restart files associated with P3. Note that for both mesh splits, solutionMigration is on because I have an initial solution on P1 that I want to migrate to P3.
The first mesh split works still successfully when buildMapping is on because there is no mapping field yet in the input restart files associated with P1.
However, Chef exits in an assert in apf.cc during the second mesh split when buildMapping is on and when the input restart files associated with P2 include already a mapping field from the first mesh split.

**Comment by KennethEJansen** [comment]
These new problems seems to stem from the change we made to ingest all fields.   We did this so that the adapt process could work with any field.

The mapping field is “special” and needs to take exception to that rule.  As Michel points out,  it is not really a “field” at all, rather it has relevance to the MOVEMENT of a field from the previous partition to the current one.  Specifically, it provides a way to take the solution BACK from the current partition to the previous one.  Under our current paradigm, it makes no sense to migrate it further since we often make these partitioning jumps at or near the limit of the process counts for a given run (e.g,, neither Chef nor our code that maps them back is capable of going forward or backward from say 8 parts to 64k parts so it is done is a series of steps (usually jumps of 64ish).

So each pass through chef produces a unique/new map for each node,part pair that tells it the node,part pair in the previous partition.  By recursively backtracking through each partition we can bring the field back to any prior partition (albeit with the requirement to visit each partition it was brought forward through).

Is this clear?

Ken

---------------------------------------------------------------- 
Kenneth E. Jansen,  Professor of Aerospace Engineering Sciences 
ECAE 161                                   OFFICE (303) 492-4359 
429 UCB                                      FAX  (303) 492-4990 
University of Colorado at Boulder          jansenke@colorado.edu 
Boulder, CO, 80309-0429        http://www.colorado.edu/aerospace 
----------------------------------------------------------------
